cmd/ztest/ztest.c

```diff
@@ -6054,6 +6049,22 @@ ztest_ddt_repair(ztest_ds_t *zd, uint64_t id)
 	blocksize = od[0].od_blocksize;
 	pattern = zs->zs_guid ^ dds.dds_guid;
 
+	/*
+	 * The block size is limited by DMU_MAX_ACCESS (64MB) which
+	 * caps the maximum transaction size.  A block size of up to
+	 * SPA_OLD_MAXBLOCKSIZE is allowed which results in a maximum
+	 * transaction size of: 128K * 200 (copies) = ~25MB
+	 *
+	 * The actual block size is checked here, rather than requested
+	 * above, because the way ztest_od_init() is implemented it does
+	 * not guarantee the block size requested will be used.
```

Comment:
This was done primary to expand the test coverage and as a little bit of cleanup.

Regarding the block size not always being as requested this was possible because the way `ztest_od_init()` was called it might reuse an existing object.  In which case you could end up using the existing objects block size.  Again for better test coverage I left that behavior unchanged and just used whatever block size we ended up with.